packages/inferno-animation/src/utils.ts

```diff
@@ -0,0 +1,206 @@
+    }
+  }
+  return oldVal
+}
+
+function _cleanStyle(node: HTMLElement) {
+  if (!node.style) {
+    // https://developer.mozilla.org/en-US/docs/Web/API/Element/removeAttribute
+    node.removeAttribute('style')
+  }
+}
+
+export function getDimensions(node: HTMLElement) {
+  var tmpDisplay = node.style.getPropertyValue('display')
+  var isDisplayNone = window.getComputedStyle(node).getPropertyValue('display') === 'none'
+  if (isDisplayNone) {
+    node.style.setProperty('display', 'block')
+  }
+
+  var tmp = node.getBoundingClientRect()
+
+  if (isDisplayNone) {
+    // node.style.display = tmpDisplay
+    node.style.setProperty('display', tmpDisplay)
+    _cleanStyle(node)
+  }
+
+  return {
+    width: tmp.width,
+    height: tmp.height
+  }
+}
+
+export function setDimensions(node: HTMLElement, width: number, height: number) {
+  node.style.width = width + 'px'
+  node.style.height = height + 'px'
+}
+
+export function clearDimensions(node: HTMLElement) {
+  node.style.width = node.style.height = ''
+}
+
+function _getMaxTransitionDuration (/* add nodes as args*/) {
+  let nrofTransitions = 0
+  let maxDuration = 0
+  for (let i=0; i < arguments.length; i++) {
+    const node = arguments[i]
+    if (!node) continue
+
+    const cs = window.getComputedStyle(node)
+    const dur = cs.getPropertyValue('transition-duration').split(',')
+    const del = cs.getPropertyValue('transition-delay').split(',')
+    const props = cs.getPropertyValue('transition-property').split(',').map((prop) => prop.trim())
+    props.forEach((prop) => {
+      if (prop[0] === '-') {
+        let tmp = prop.split('-').splice(2).join('-')
+        if (props.indexOf(tmp) >= 0) {
+          nrofTransitions--
+        }
+      }
+    })
+    const animTimeout = dur.map((v, index) => parseFloat(v) + parseFloat(del[index])).reduce((prev, curr) => prev > curr ? prev : curr, 0)
+
+    nrofTransitions += dur.length
+    if (animTimeout > maxDuration) {
+      maxDuration = animTimeout
+    }
+  }
+
+  return {
+    nrofTransitions: nrofTransitions,
+    maxDuration: maxDuration
+  }
+}
+
+function whichTransitionEvent(){
+  var el = document.createElement('fakeelement');
+  var transitions = {
+    'transition':'transitionend',
+    'OTransition':'oTransitionEnd',
+    'MozTransition':'transitionend',
+    'WebkitTransition':'webkitTransitionEnd'
+  }
+
+  for(let t in transitions){
+      if( el.style[t] !== undefined ){
+          return transitions[t];
+      }
+  }
+}
+var transitionEndName: string;
+
+/**
+ * You need to pass the root element and ALL animated children that have transitions,
+ * if there are any,  so the timeout is set to the longest duration. Otherwise there
+ * will be animations that fail to complete before the timeout is triggered.
+ */
+export function registerTransitionListener(nodes: HTMLElement[], callback: Function, noTimeout: boolean = false) {
+  // I am doing this lazily because there where issues with document being undefined
```

Comment:
const hasDocumentAvailable: boolean = typeof document !== 'undefined';

Reply:
Will investigate

Reply:
FIXED in 695fe367e